src/main.cpp

```diff
@@ -140,14 +142,20 @@ int load_conf_file(char *file_path)
             char *current_test;
             char *tests = strtok(NULL, NEWLINE_DELIMITER);
             while ((current_test = strtok_r(tests, ",", &tests))) {
-                vector_append(&tests_to_run, strtol(current_test, NULL, 16));
+                unsigned long value = strtoul(current_test, NULL, 16);
+                if (value < kernel_api_tests_size) {
+                    tests_to_run.set(value);
+                }
             }
         }
         if (strcmp("tests-exclude", current_key) == 0) {
             char *current_test;
             char *tests = strtok(NULL, NEWLINE_DELIMITER);
             while ((current_test = strtok_r(tests, ",", &tests))) {
-                vector_append(&tests_exclude, strtol(current_test, NULL, 16));
+                unsigned long value = strtoul(current_test, NULL, 16);
+                if (value < kernel_api_tests_size) {
```

Comment:
Would `tests_to_exclude[value]` be possible (and preferable) here?

Reply:
No, what you suggest is outside the limit of "size" which will cause access violation. Unless you're referring to one line below. Both `.set()` and `.reset()` are equivalent to using `var[value] = true/false;`, except I choose to use those bitset's functions directly.

If above are not what you're suggesting. Please clarify your question more clearly.

Reply:
Continued conversation: https://github.com/Cxbx-Reloaded/xbox_kernel_test_suite/pull/105#discussion_r1526942957